File: Part8/Python/FunctionConcepts/FunctionConcepts_11.py
# Recursive function call to compute factorial
def factorial ( FactorialNumber_Par ):
    if FactorialNumber_Par <= 1:
        return 1
    
    return FactorialNumber_Par * factorial ( FactorialNumber_Par - 1 )

def main():
    factorialResult = factorial ( FactorialNumber_Par=10 )
    print ( f"Factorial of 10 is: {factorialResult}" )

if __name__ == "__main__":
    main()

File: Part8/C#/FunctionConcepts/FunctionConcepts_11.cs
// Recursive function call to compute factorial
using System;

class Program
{
    static int Factorial ( int FactorialNumber_Par )
    {
        if ( FactorialNumber_Par <= 1 )
        {
            return 1;
        }
        
        return FactorialNumber_Par * Factorial ( FactorialNumber_Par - 1 );
    }

    static void Main ()
    {
        int factorialResult = Factorial ( FactorialNumber_Par: 10 );
        Console.WriteLine ( "Factorial of 10 is: {0}", factorialResult );
    }
}
